src/main/java/com/corundumstudio/socketio/handler/ClientsBox.java

```diff
@@ -15,19 +15,45 @@
  */
 package com.corundumstudio.socketio.handler;
 
+import com.corundumstudio.socketio.HandshakeData;
 import io.netty.channel.Channel;
 import io.netty.util.internal.PlatformDependent;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Map;
 import java.util.UUID;
-
-import com.corundumstudio.socketio.HandshakeData;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
 
 public class ClientsBox {
+    private static final Logger log = LoggerFactory.getLogger(ClientsBox.class);
 
     private final Map<UUID, ClientHead> uuid2clients = PlatformDependent.newConcurrentHashMap();
     private final Map<Channel, ClientHead> channel2clients = PlatformDependent.newConcurrentHashMap();
 
+    private final ScheduledExecutorService scheduledExecutorService = Executors.newSingleThreadScheduledExecutor();
+
+    public ClientsBox() {
+        scheduledExecutorService.scheduleWithFixedDelay(() -> {
```

Comment:
Why do we need a periodical task for this and schedule this check after client connection?

Reply:
I am afraid there would be bugs in future commit that cause this is not dereferenced after client disconnected

Reply:
like the one this PR fix, we cannot confirm there would not be any bugs alike in the future. but we should not let this kind of bug cause Memory Leak. We should actively detect that, remove that and warn it by log

Reply:
this logic should be implemented in com.corundumstudio.socketio.handler.ClientHead#onChannelDisconnect() method without ScheduledExecutorService usage.

Reply:
> this logic should be implemented in com.corundumstudio.socketio.handler.ClientHead#onChannelDisconnect() method without ScheduledExecutorService usage.

but like the bug this time, `onChannelDisconnect` is called indeed(due to scheduled ping timeout), but the `namespaceClients` does not contain the client which throw WebSocketException, which result in the `uuid2clients` does not remove it.

Reply:
You're checking entry.getValue().isConnected() but it can be set only in onChannelDisconnect() handler. I suggest to call clientsBox.removeClient() in this method

Reply:
@HashZhang